lib/container_server_test.go

```diff
@@ -0,0 +1,1066 @@
+
+			// When
+			err := sut.LoadSandbox("id")
+
+			// Then
+			Expect(err).NotTo(BeNil())
+		})
+
+		It("should fail with failing container run directory", func() {
+			// Given
+			gomock.InOrder(
+				storeMock.EXPECT().
+					FromContainerDirectory(gomock.Any(), gomock.Any()).
+					Return(testManifest, nil),
+				storeMock.EXPECT().ContainerRunDirectory(gomock.Any()).
+					Return("", t.TestError),
+			)
+
+			// When
+			err := sut.LoadSandbox("id")
+
+			// Then
+			Expect(err).NotTo(BeNil())
+		})
+
+		It("should fail with invalid namespace options", func() {
+			// Given
+			manifest := bytes.Replace(testManifest,
+				[]byte(`"io.kubernetes.cri-o.NamespaceOptions": "{}",`),
+				[]byte(`"io.kubernetes.cri-o.NamespaceOptions": "",`), 1,
+			)
+			gomock.InOrder(
+				storeMock.EXPECT().
+					FromContainerDirectory(gomock.Any(), gomock.Any()).
+					Return(manifest, nil),
+			)
+
+			// When
+			err := sut.LoadSandbox("id")
+
+			// Then
+			Expect(err).NotTo(BeNil())
+		})
+
+		It("should fail with invalid port mappings", func() {
+			// Given
+			manifest := bytes.Replace(testManifest,
+				[]byte(`"io.kubernetes.cri-o.PortMappings": "[]",`),
+				[]byte(`"io.kubernetes.cri-o.PortMappings": "{}",`), 1,
+			)
+			gomock.InOrder(
+				storeMock.EXPECT().
+					FromContainerDirectory(gomock.Any(), gomock.Any()).
+					Return(manifest, nil),
+			)
+
+			// When
+			err := sut.LoadSandbox("id")
+
+			// Then
+			Expect(err).NotTo(BeNil())
+		})
+
+		It("should fail with invalid kube annotations", func() {
+			// Given
+			manifest := bytes.Replace(testManifest,
+				[]byte(`"io.kubernetes.cri-o.Annotations": "{}",`),
+				[]byte(`"io.kubernetes.cri-o.Annotations": "",`), 1,
+			)
+			gomock.InOrder(
+				storeMock.EXPECT().
+					FromContainerDirectory(gomock.Any(), gomock.Any()).
+					Return(manifest, nil),
+			)
+
+			// When
+			err := sut.LoadSandbox("id")
+
+			// Then
+			Expect(err).NotTo(BeNil())
+		})
+
+		It("should fail with invalid metadata", func() {
+			// Given
+			manifest := bytes.Replace(testManifest,
+				[]byte(`"io.kubernetes.cri-o.Metadata": "{}",`),
+				[]byte(`"io.kubernetes.cri-o.Metadata": "",`), 1,
+			)
+			gomock.InOrder(
+				storeMock.EXPECT().
+					FromContainerDirectory(gomock.Any(), gomock.Any()).
+					Return(manifest, nil),
+			)
+
+			// When
+			err := sut.LoadSandbox("id")
+
+			// Then
+			Expect(err).NotTo(BeNil())
+		})
+
+		It("should fail with invalid labels", func() {
+			// Given
+			manifest := bytes.Replace(testManifest,
+				[]byte(`"io.kubernetes.cri-o.Labels": "{}",`),
+				[]byte(`"io.kubernetes.cri-o.Labels": "",`), 1,
+			)
+			gomock.InOrder(
+				storeMock.EXPECT().
+					FromContainerDirectory(gomock.Any(), gomock.Any()).
+					Return(manifest, nil),
+			)
+
+			// When
+			err := sut.LoadSandbox("id")
+
+			// Then
+			Expect(err).NotTo(BeNil())
+		})
+
+		// TODO(sgrunert): uncomment when the following issue is resolved:
+		// https://github.com/kubernetes-sigs/cri-o/issues/1987
```

Comment:
#1987 was fixed. This test can be uncommented now.

Reply:
Ah I verified that it is not resolved yet.